server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestRefreshAction.java

```diff
@@ -57,14 +53,6 @@ public String getName() {
     public RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
         RefreshRequest refreshRequest = new RefreshRequest(Strings.splitStringByCommaToArray(request.param("index")));
         refreshRequest.indicesOptions(IndicesOptions.fromRequest(request, refreshRequest.indicesOptions()));
-        return channel -> client.admin().indices().refresh(refreshRequest, new RestBuilderListener<RefreshResponse>(channel) {
-            @Override
-            public RestResponse buildResponse(RefreshResponse response, XContentBuilder builder) throws Exception {
-                builder.startObject();
-                response.toXContent(builder, request);
-                builder.endObject();
-                return new BytesRestResponse(response.getStatus(), builder);
-            }
-        });
+        return channel -> client.admin().indices().refresh(refreshRequest, new RestToXContentListener<>(channel));
```

Comment:
we are losing the status code from refresh response here. You need to override `getStatus` in `RestToXContentListener` and take the status from the response.